File: justsolomon/01-arrays-and-strings/reverse-sentence.py
def reverseSentence(A):
	def reverseWord(start, end):
		if start > end:
			return
	
		while start < end:
			A[start], A[end] = A[end], A[start]
			
			start += 1
			end -= 1
	
	
	currentStart = 0
	length = len(A)
	
	reverseWord(0, length - 1)
	
	for i in range(length):
		if A[i] == " ":
			reverseWord(currentStart, i - 1)
			currentStart = i + 1
		elif i == length - 1:
			reverseWord(currentStart, i)
	
	return A

File: arrays-and-strings/reverseSentence.js
var A = ['t','h','i','s',' ','i','s',' ','g','o','o','d']
reverseSentence(A)
// ['g','o','o','d',' ','i','s',' ','t','h','i','s']

function reverseSentence(arr){

    arr = arr.join('');
    arr = arr.split(' ');

    var start = 0;
    var stop = arr.length - 1;


    while (start < stop){
        var temp = arr[start];
        arr[start] = arr[stop];
        arr[stop] = temp;
        start++;
        stop--;
    }

    for(var i = 0; i < arr.length; i++){
        arr[i] = arr[i].split('');
    }

    arr = arr.join(', ,')

    console.log(arr)
}

